common/namespace/nsregistry/registry.go

```diff
@@ -220,52 +218,37 @@ func (r *registry) GetPingChecks() []pingable.Check {
 			Timeout: 10 * time.Second,
 			Ping: func() []pingable.Pingable {
 				// just checking if we can acquire the lock
-				r.cacheLock.Lock()
+				r.registryLock.Lock()
 				// nolint:staticcheck
-				r.cacheLock.Unlock()
+				r.registryLock.Unlock()
 				return nil
 			},
 			MetricsName: metrics.DDNamespaceRegistryLockLatency.Name(),
 		},
 	}
 }
 
-func (r *registry) getAllNamespace() map[namespace.ID]*namespace.Namespace {
-	r.cacheLock.RLock()
-	defer r.cacheLock.RUnlock()
+func (r *registry) getAllNamespace() []*namespace.Namespace {
+	r.registryLock.RLock()
+	defer r.registryLock.RUnlock()
 	return r.getAllNamespaceLocked()
 }
 
-func (r *registry) getAllNamespaceLocked() map[namespace.ID]*namespace.Namespace {
-	result := make(map[namespace.ID]*namespace.Namespace)
-
-	ite := r.cacheByID.Iterator()
-	defer ite.Close()
-
-	for ite.HasNext() {
-		entry := ite.Next()
-		id, ok := entry.Key().(namespace.ID)
-		if !ok {
-			// This can be stripped once Cache is fully typed with generics
-			r.logger.Error("Got to theoretically impossible branch: key in cache is not namespace.ID")
-			return nil
-		}
-		aResult, ok := entry.Value().(*namespace.Namespace)
-		if !ok {
-			// This can be stripped once Cache is fully typed with generics
-			r.logger.Error("Got to theoretically impossible branch: value in cache is not namespace.Namespace")
-			return nil
-		}
-		result[id] = aResult
+func (r *registry) getAllNamespaceLocked() []*namespace.Namespace {
+	result := make([]*namespace.Namespace, len(r.idToNamespace))
+	i := 0
+	for _, value := range r.idToNamespace {
+		result[i] = value
+		i++
 	}
 	return result
```

Comment:
```suggestion
	return expmaps.Values(r.idToNamespace)
```
where `expmaps` is `import expmaps "golang.org/x/exp/maps"`